olp-cpp-sdk-core/tests/logging/MessageFormatterTest.cpp

```diff
@@ -0,0 +1,501 @@
+    EXPECT_EQ( "[ERROR]", map[ static_cast< std::size_t >( Level::Error ) ] );
+    EXPECT_EQ( "[FATAL]", map[ static_cast< std::size_t >( Level::Fatal ) ] );
+}
+
+TEST( MessageFormatterTest, DefaultConstructor )
+{
+    MessageFormatter formatter;
+    EXPECT_TRUE( formatter.getElements( ).empty( ) );
+    EXPECT_EQ( MessageFormatter::defaultLevelNameMap( ), formatter.getLevelNameMap( ) );
+    EXPECT_EQ( MessageFormatter::Timezone::Local, formatter.getTimezone( ) );
+}
+
+TEST( MessageFormatterTest, Constructor )
+{
+    std::vector< MessageFormatter::Element > elements
+        = {MessageFormatter::Element( MessageFormatter::ElementType::String, "LOG: " ),
+           MessageFormatter::Element( MessageFormatter::ElementType::Level, "%s " ),
+           MessageFormatter::Element( MessageFormatter::ElementType::Tag, "%s - " ),
+           MessageFormatter::Element( MessageFormatter::ElementType::File, "%30s:", -30 ),
+           MessageFormatter::Element( MessageFormatter::ElementType::Line, "%5u " ),
+           MessageFormatter::Element( MessageFormatter::ElementType::Time, "[%H:%M] " ),
+           MessageFormatter::Element( MessageFormatter::ElementType::Message )};
+
+    MessageFormatter::LevelNameMap levelNameMap
+        = {{{"Trace"}, {"Debug"}, {"Info"}, {"Warning"}, {"Error"}, {"Fatal"}}};
+
+    MessageFormatter formatter( elements, levelNameMap, MessageFormatter::Timezone::Utc );
+    EXPECT_EQ( elements, formatter.getElements( ) );
+    EXPECT_EQ( levelNameMap, formatter.getLevelNameMap( ) );
+    EXPECT_EQ( MessageFormatter::Timezone::Utc, formatter.getTimezone( ) );
+}
+
+TEST( MessageFormatterTest, CreateDefault )
+{
+    MessageFormatter formatter = MessageFormatter::createDefault( );
+    EXPECT_FALSE( formatter.getElements( ).empty( ) );
+    EXPECT_EQ( MessageFormatter::defaultLevelNameMap( ), formatter.getLevelNameMap( ) );
+    EXPECT_EQ( MessageFormatter::Timezone::Local, formatter.getTimezone( ) );
+}
+
+TEST( MessageFormatterTest, SetElements )
+{
+    MessageFormatter formatter;
+
+    std::vector< MessageFormatter::Element > elements
+        = {MessageFormatter::Element( MessageFormatter::ElementType::String, "LOG: " ),
+           MessageFormatter::Element( MessageFormatter::ElementType::Level, "%s " ),
+           MessageFormatter::Element( MessageFormatter::ElementType::Tag, "%s - " ),
+           MessageFormatter::Element( MessageFormatter::ElementType::File, "%30s:", -30 ),
+           MessageFormatter::Element( MessageFormatter::ElementType::Line, "%5u " ),
+           MessageFormatter::Element( MessageFormatter::ElementType::Time, "[%H:%M] " ),
+           MessageFormatter::Element( MessageFormatter::ElementType::Message )};
+
+    EXPECT_NE( elements, formatter.getElements( ) );
+    formatter.setElements( elements );
+    EXPECT_EQ( elements, formatter.getElements( ) );
+}
+
+TEST( MessageFormatterTest, SetLevelNameMap )
+{
+    MessageFormatter formatter;
+
+    MessageFormatter::LevelNameMap levelNameMap
```

Comment:
```suggestion
    MessageFormatter::LevelNameMap levelNameMap
```
Same here - consider to move this map into common place (anonymous namespace on top), as it is used in `MessageFormatterTest.Constructor` test.